File: cpp/armstrong-numbers/armstrong_numbers.cpp
#include "armstrong_numbers.h"
#include <cmath>

namespace armstrong_numbers {

    bool is_armstrong_number(int number) {
        int sum = 0;
        int power = (int) std::log10(number) + 1;
        int remainder = number;
        while (remainder) {
            int digit = remainder % 10;
            sum += (int) std::pow(digit, power);
            remainder = remainder / 10;
        }
        return number == sum;
    }

}  // namespace armstrong_numbers

File: javascript/armstrong-numbers/armstrong-numbers.js
export const isArmstrongNumber = (n) => {
  const power = Math.floor(Math.log10(n) + 1);
  let [sum, remainder] = [0, n];
  while (remainder) {
    sum += Math.pow(remainder % 10, power);
    remainder = Math.floor(remainder / 10);
  }
  return sum == n;
};
